Compilation failing on Windows

The `build.bat` script used to compile in Windows is not working properly. It seems we have a piece of code missing from one of the include libraries:

```
F:\onyx_src_benji\shared\include\bh.h(1930): error C2039: 'st_mtim': is not a member of 'stat'
C:\Program Files (x86)\Windows Kits\10\include\10.0.22621.0\ucrt\sys/stat.h(87): note: see declaration of 'stat'
F:\onyx_src_benji\shared\include\bh.h(1930): error C2198: 'timespec_to_ms': too few arguments for call
F:\onyx_src_benji\shared\include\bh.h(1931): error C2039: 'st_atim': is not a member of 'stat'
C:\Program Files (x86)\Windows Kits\10\include\10.0.22621.0\ucrt\sys/stat.h(87): note: see declaration of 'stat'
F:\onyx_src_benji\shared\include\bh.h(1931): error C2198: 'timespec_to_ms': too few arguments for call
F:\onyx_src_benji\shared\include\bh.h(1932): error C2039: 'st_ctim': is not a member of 'stat'
C:\Program Files (x86)\Windows Kits\10\include\10.0.22621.0\ucrt\sys/stat.h(87): note: see declaration of 'stat'
F:\onyx_src_benji\shared\include\bh.h(1932): error C2198: 'timespec_to_ms': too few arguments for call
```

In particular, it seems we're missing a check for `_BH_WINDOWS` here:
```
#if defined(_BH_DARWIN)
    // Apple just has to be different.
    out->modified_time = timespec_to_ms(s.st_mtimespec);
    out->accessed_time = timespec_to_ms(s.st_atimespec);
    out->change_time   = timespec_to_ms(s.st_ctimespec);
#else
    out->modified_time = timespec_to_ms(s.st_mtim);
    out->accessed_time = timespec_to_ms(s.st_atim);
    out->change_time   = timespec_to_ms(s.st_ctim);
#endif
```

Additionally, the batch script does **not** exit with an error code after compilation fails, so CI succeeds. The following pipeline passed, even though it couldn't build an `onyx` executable: https://github.com/onyx-lang/onyx/actions/runs/7946470284/job/21694237022

"Build Onyx for windows-amd64-" also tells us there was a problem copying the executable. Adding a few `echo`s helps us diagnose the problem:
```
Trying to copy onyx.exe to dist...
The system cannot find the file specified.
!! Seems there wasn't any onyx.exe!
```

--- a/build.bat
+++ b/build.bat
@@ -17,12 +17,23 @@ set FLAGS=%FLAGS% "/I%PWD%\shared\include" /DONYX_RUNTIME_LIBRARY=wasmer
 rc.exe misc/icon_resource.rc
 cl.exe %FLAGS% /Icompiler/include /std:c17 /TC %SOURCE_FILES% /link /IGNORE:4217 %LINK_OPTIONS% /DEBUG /OUT:onyx.exe /incremental:no /opt:ref /subsystem:console misc\icon_resource.res
 
+REM Don't continue if we had compilation errors. This prevents CI to succeed.
+if %ERRORLEVEL% neq 0 (
+    echo Compiler compilation failed.
+    exit /b %ERRORLEVEL%
+)
+
 del *.pdb > NUL 2> NUL
 del *.ilk > NUL 2> NUL
 del *.obj > NUL 2> NUL
 del misc\icon_resource.res
 
 cl /MT /std:c17 /TC /I compiler/include /I shared/include /D_USRDLL /D_WINDLL runtime\onyx_runtime.c /link /DLL ws2_32.lib bcrypt.lib Synchronization.lib kernel32.lib /OUT:onyx_runtime.dll
+
+if %ERRORLEVEL% neq 0 (
+    echo Onyx runtime compilation failed.
+    exit /b %ERRORLEVEL%
+)
 
 del onyx_runtime.obj
 del onyx_runtime.lib

--- a/shared/include/bh.h
+++ b/shared/include/bh.h
@@ -1926,6 +1926,10 @@ b32 bh_file_stat(char const* filename, bh_file_stats* out) {
     out->modified_time = timespec_to_ms(s.st_mtimespec);
     out->accessed_time = timespec_to_ms(s.st_atimespec);
     out->change_time   = timespec_to_ms(s.st_ctimespec);
+#elif defined(_BH_WINDOWS)
+    out->modified_time = s.st_mtime * 1000;
+    out->accessed_time = s.st_atime * 1000;
+    out->change_time   = s.st_ctime * 1000;
 #else
     out->modified_time = timespec_to_ms(s.st_mtim);
     out->accessed_time = timespec_to_ms(s.st_atim);

